src/constructs/autoscaling/user-data.ts

```diff
@@ -1,13 +1,77 @@
 import type { S3DownloadOptions } from "@aws-cdk/aws-ec2";
 import { UserData } from "@aws-cdk/aws-ec2";
+import { Bucket } from "@aws-cdk/aws-s3";
+import type { GuStack } from "../core";
+
+export interface GuUserDataS3DistributableProps {
```

Comment:
I think it's worth adding some basic docs for these interfaces now so that we don't forget?

Reply:
Done.